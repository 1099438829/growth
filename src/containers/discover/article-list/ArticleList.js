import React, { Component, PropTypes } from 'react';
import { ScrollView } from 'react-native';
import AppStyle from '../../../theme/styles';
import Api from '../../../utils/api';
import Dialog from '../../../components/dialog';
import SimpleListItem from '../SimpleListItem';
import Launch from '../Launch';

class ArticleList extends Component {
  static componentName = 'ArticleList';

  static propTypes = {
    dialogContent: PropTypes.string,
  };

  static defaultProps = {
    dialogContent: '',
  };

  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      rowData: [],
    };
  }

  componentDidMount() {
    Api.get(Api.ARTICLE_LIST)
      .then(response => this.setState({
        loading: false,
        rowData: Array.from(new Array(response.data.content.length))
          .map((val, index) => (response.data.content[index])) }));
  }

  render() {
    const rows = this.state.rowData.map((val, index) => (
      <SimpleListItem text={val.title} click={() => Launch.articleDetail(val.path)} key={'key'.concat(index)} />
    ));
    return (
      <ScrollView style={AppStyle.detailBasisStyle}>
        <Dialog show={this.state.loading} content={this.props.dialogContent} />
        {rows}
      </ScrollView>);
  }
}
export default ArticleList;
