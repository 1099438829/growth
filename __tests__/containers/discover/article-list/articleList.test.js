import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import ArticleList from '../../../../src/containers/discover/article-list/ArticleList';

it('renders correclty', () => {
  const articleList = renderer.create(
    <ArticleList />,
  );
  const articleListJson = articleList.toJSON();
  expect(articleListJson).toMatchSnapshot();
});
