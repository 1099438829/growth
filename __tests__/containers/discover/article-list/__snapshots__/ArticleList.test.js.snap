// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty 1`] = `null`;
